src/Page/CKANRegistryPage.php

```diff
@@ -134,4 +134,9 @@ public function getSettingsFields()
 
         return $fields;
     }
+
+    public function getCKANClientConfig()
```

Comment:
This feels more like a controller method than a model method to me. I think also it could use a doc block explaining what it's for and also some basic scaffolding for future use because I suspect the return below is temporary:

```php
$config = [];
$this->extend('updateCKANClientConfig', $config);
return $config;
```

You could either return `json_encode($config)` or define some `$casting` to cast it internally as JSON

Reply:
Good points. It was very much an MVP effort to get off the ground :>
The plan is certainly to add hooks, docblocks, json_encode.

The controller part I'm not 100% convinced of, but nor am I against it really. We would need to introduce the controller just for it - which may not always be used (e.g. `dnadesign/silverstripe-elemental` context). This is why I added it here :)

Reply:
Yes sorry the first point was simply a statement of necessity.
The second point is relevant because the relationship for the DataStore would be on the Element, not the Page. If this is true (i.e. someone has created a `CKANResourceElement`) then the rendering has no context of the page controller, and likewise the page controller has no concept of a CKAN data resource.

Which begs the question... should this config (which is a representation of the data model) not live on the `SilverStripe\CKANRegistry\Model\Resource` instead?

Reply:
>should this config (which is a representation of the data model) not live on the SilverStripe\CKANRegistry\Model\Resource instead?

Makes sense - that'd future proof it for conversion to an element as well